### PR TITLE
Cleanup tools/wpt/tests

### LINKED ISSUE #9056: Get rid of infra tests marked "system_dependent"
In https://github.com/w3c/web-platform-tests/commit/b4b9ec466d8f719da9876c00afcf709529f1e4a6 I added a bunch of markers to a number of tests, including "system_dependent". We should aim to get rid of these markers, either by getting rid of the system dependence or by marking them as skipped unless whatever condition they require is met.

### PR DESCRIPTION
**Pull request by gsnedders**
Fixes #9056.

Having done a load of this while investigating #9058 to try and get the tests to run on Windows, we may as well land this.

This makes it possible to run all of the wpt tests locally, and not rely on Travis.

---
This change is [Reviewable](https://reviewable.io/reviews/w3c/web-platform-tests/9069)

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Build PASSED

Started: 2018-01-17 21:16:53
Finished: 2018-01-17 21:30:51





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/22936)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/330054648)

File: tools/wpt/tests/test_wpt.py [hunk 1]
```diff
@@ -13,6 +14,43 @@
 pytestmark = pytest.mark.skipif(os.name == "nt",
                                 reason="Tests currently don't work on Windows for path reasons")
 
+def is_port_8000_in_use():
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    try:
+        s.bind(("127.0.0.1", 8000))
+    except socket.error as e:
+        if e.errno == 98:
```
Comment (jgraham) [inline tools/wpt/tests/test_wpt.py]: Use the named constant in the `errno` module.

File: tools/wpt/tests/test_wpt.py [hunk 2]
```diff
@@ -27,19 +65,21 @@ def test_help():
...
 
 @pytest.mark.slow
-@pytest.mark.system_dependent
 @pytest.mark.remote_network
-def test_run_firefox():
+def test_run_firefox(manifest_dir):
     # TODO: It seems like there's a bug in argparse that makes this argument order required
     # should try to work around that
+    if is_port_8000_in_use():
+        pytest.skip("port 8000 already in use")
```
Comment (jgraham) [inline tools/wpt/tests/test_wpt.py]: OK, I guess, but probably people aren't going to notice that they didn't run all the tests.

**Review by jgraham — CHANGES_REQUESTED** [review]
Just one change, otehrwise this looks good, thanks.

File: tools/wpt/tests/test_wpt.py (hunk 1 above)
Comment (gsnedders) [reply tools/wpt/tests/test_wpt.py]: Ah, I couldn't find a constant for it anywhere!

**Comment by gsnedders** [comment]
ping @jgraham

**Comment by gsnedders** [comment]
@jgraham I'd really like this to land so that it's possible to run *some* tests with decent coverage of wptrunner locally.

**Review by jgraham — APPROVED** [review]

**Pull request merged**